parquet-column/src/main/java/org/apache/parquet/column/statistics/Statistics.java

```diff
@@ -175,9 +217,48 @@ public void mergeStatistics(Statistics stats) {
    */
   abstract public void setMinMaxFromBytes(byte[] minBytes, byte[] maxBytes);
 
+  /**
+   * Returns the min value in the statistics. The java natural order of the returned type defined by {@link
+   * T#compareTo(Object)} might not be the proper one. For example, UINT_32 requires unsigned comparison instead of the
+   * natural signed one. Use {@link #compareToMin(Comparable)} or the comparator returned by {@link #comparator()} to
+   * always get the proper ordering.
+   */
   abstract public T genericGetMin();
+
+  /**
+   * Returns the max value in the statistics. The java natural order of the returned type defined by {@link
+   * T#compareTo(Object)} might not be the proper one. For example, UINT_32 requires unsigned comparison instead of the
+   * natural signed one. Use {@link #compareToMax(Comparable)} or the comparator returned by {@link #comparator()} to
+   * always get the proper ordering.
+   */
   abstract public T genericGetMax();
 
+  /**
+   * Returns the comparator to be used to compare two generic values in the proper way (for example, unsigned comparison
+   * for UINT_32).
+   */
+  public final PrimitiveComparator<T> comparator() {
+    return comparator;
+  }
+
+  /**
+   * Compares the specified value to min in the proper way.
+   *
+   * @see Comparable#compareTo(Object)
+   */
+  public final int compareToMin(T value) {
+    return comparator.compare(genericGetMin(), value);
+  }
+
+  /**
+   * Compares the specified value to max in the proper way.
+   *
+   * @see Comparable#compareTo(Object)
+   */
+  public final int compareToMax(T value) {
+    return comparator.compare(genericGetMax(), value);
```

Comment:
From reading the tests, this method gives no indication whether the order of comparison is `max, value` or `value, max`. That's going to lead to bugs later, so I think there are two options:
1. Remove this and encourage using `comparator().compare(value, stats.max())` with a primitive `max` accessor
2. Change this to `isGreaterThanMax(T value)` that makes the comparison order obvious

Reply:
Actually, I don't like either. The first one is a strange pattern to get the comparator and the max value from the same object and then invoke compare. The second one does not answer whether it is equal or less. (In filters we need to implement all `<`, `<=`, `>`, `>=` and `==` comparisons.)
What do you think about renaming to `compareMaxToValue(T value)` and document the comparison properly (`max, value`)?

Reply:
I like this.